changelog/target.md

```diff
@@ -0,0 +1,72 @@
+Add `-target=<triple>` for operating system, c, and c++ runtime cross compilation
+
+The command line switch `-target=<triple>` can be used to specify the target the
+compiler should produce code for. The format for `<triple>` is `<arch>-[<vendor>-]<os>[-<cenv>[-<cppenv]]`. Specific values of the specifiers are 
+shown in the tables below.
+
+`<arch>` 
+---------
+`<arch>` is an Architecture bitness, optionally followed by instruction set (which duplicate the functionality of the `-mcpu` flag)
+| `<arch>`  | Architecture bitness |
+| ---------- |-------------| 
+| `x86_64`  |  64 bit |
+| `x64`        |  64 bit |
+| `x86`        |  32 bit |
+| `x32`        |  64bit with 32 bit pointers |
+
+| subarch features | equivalent `-mcpu` flag |
+| ---------- |-------------| 
+| `+sse2`    | `baseline` |
+| `+avx`      |  `avx` |
+| `+avx2`    |  `avx2` |
+
+The architecture and subarch features are concatenated, so `x86_64+avx2` is 64bit with `avx2` instructions.
+
+
+`<vendor>` 
+---------
+`<vendor>` is always ignored, but supported for easier interoperability with other compilers that report and consume target triples.
+
+`<os>`
+------
+
+`<os>` is the operating system specifier. It may be optionally followed by a version number as is `darwin20.3.0` or `freebsd12`. For Freebsd, this sets the corresponding predefined `version` identifier, e.g. `freebsd12` sets `version(FreeBSD12)`. For other operating systems this is for easier interoperability with other compilers.
+
+| `<os>` | Operating system | 
+| ---------- |-------------| 
+| `freestanding` | No operating system |
+| `darwin`  or `osx` | MacOS, iOS, tvOS | 
+| `dragonfly` | DragonflyBSD | 
+| `freebsd` | FreeBSD |
+| `openbsd` | OpenBSD | 
+| `linux` | Linux |
+| `solaris` | Solaris |
+| `windows` | Windows |
+
+`<cenv>`
+---------
+`<cenv>` is the C runtime environment. This specifier is optional. For MacOS the C runtime environment is assumed to be the system default.
```

Comment:
> For MacOS the C runtime environment is assumed to be the system default.

Doesn't this apply to all platforms except Linux and Windows?

Reply:
Hmm, this needs to be reworded. I meant that there is no way to set a different one. What is the name of the C runtime environment on macos?

Reply:
There is no name, but you can use the string to set what baseline version of darwin you want to target.

Reply:
os version is part of the os field not the env field. e.g. my `clang --version` reports `arm64-apple-darwin20.3.0`

Reply:
reworded to make the intention more clear

Reply:
I don't see any difference. Did you just add `always`? Why are you calling out just macOS? It should apply to FreeBSD, OpenBSD, DragonFly BSD and Solaris as well. Why don't you word it so that you mention that it's only possible to change this on Windows and Linux.